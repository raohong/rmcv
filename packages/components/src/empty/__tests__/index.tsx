import { render, screen } from '@testing-library/react';
import { Empty } from '..';

const testId = 'empty';

it('render correctly', () => {
  const tree = render(<Empty />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with description', () => {
  render(<Empty data-testid={testId} description='desc' />);

  expect(screen.getByText('desc')).not.toBeNull();
});

it('render with image', () => {
  render(<Empty data-testid={testId} image={<span>image</span>} />);

  expect(screen.getByText('image')).not.toBeNull();
  expect(screen.getByTestId(testId).querySelector('svg')).not.toBeInTheDocument();
});
