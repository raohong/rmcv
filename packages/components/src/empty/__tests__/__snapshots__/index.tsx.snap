// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  <div
    class="rmcvant-empty"
  >
    <div
      class="rmcvant-empty-image"
    >
      <svg
        height="160"
        width="160"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <lineargradient
            id="rmc-vant-empty-default__a"
            x1="64.022%"
            x2="64.022%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
              stop-opacity="0.5"
            />
            <stop
              offset="100%"
              stop-color="#F2F3F5"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__b"
            x1="64.022%"
            x2="64.022%"
            y1="96.956%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#F2F3F5"
              stop-opacity="0.3"
            />
            <stop
              offset="100%"
              stop-color="#F2F3F5"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__c"
            x1="64.022%"
            x2="64.022%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
              stop-opacity="0.5"
            />
            <stop
              offset="100%"
              stop-color="#F2F3F5"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__d"
            x1="64.022%"
            x2="64.022%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
              stop-opacity="0.5"
            />
            <stop
              offset="100%"
              stop-color="#F2F3F5"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__f"
            x1="50%"
            x2="50%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#F2F3F5"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__g"
            x1="50%"
            x2="50%"
            y1="100%"
            y2="0%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__i"
            x1="50%"
            x2="50%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#F2F3F5"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__j"
            x1="50%"
            x2="50%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
              stop-opacity="0.635"
            />
            <stop
              offset="100%"
              stop-color="#FFF"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__l"
            x1="0%"
            x2="99.63%"
            y1="21.18%"
            y2="78.607%"
          >
            <stop
              offset="0%"
              stop-color="#F2F3F5"
            />
            <stop
              offset="99.926%"
              stop-color="#DCDEE0"
            />
            <stop
              offset="100%"
              stop-color="#DCDEE0"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__m"
            x1="3.183%"
            x2="72.792%"
            y1="50%"
            y2="51.223%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
            />
            <stop
              offset="100%"
              stop-color="#F7F8FA"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__n"
            x1="3.183%"
            x2="72.792%"
            y1="50%"
            y2="52.455%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
            />
            <stop
              offset="100%"
              stop-color="#F7F8FA"
            />
          </lineargradient>
          <lineargradient
            id="rmc-vant-empty-default__o"
            x1="3.183%"
            x2="72.792%"
            y1="50%"
            y2="51.273%"
          >
            <stop
              offset="0%"
              stop-color="#FFF"
            />
            <stop
              offset="100%"
              stop-color="#F7F8FA"
            />
          </lineargradient>
          <radialgradient
            cx="50%"
            cy="54.421%"
            fx="50%"
            fy="54.421%"
            gradientTransform="matrix(-.16318 0 0 -.32617 .582 .722)"
            id="rmc-vant-empty-default__e"
            r="297.427%"
          >
            <stop
              offset="0%"
              stop-color="#EBEDF0"
            />
            <stop
              offset="100%"
              stop-color="#F2F3F5"
              stop-opacity="0.3"
            />
          </radialgradient>
          <rect
            height="50"
            id="rmc-vant-empty-default__h"
            rx="2"
            width="78"
            x="0"
            y="0"
          />
        </defs>
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M2 131V53h20v20h14v58H2z"
            fill="url(#rmc-vant-empty-default__a)"
            opacity="0.8"
            transform="matrix(-1 0 0 1 38 0)"
          />
          <path
            d="M35.5 15a8.504 8.504 0 018.13 6.009l.12-.005L44 21a8 8 0 11-1 15.938V37H29v-.174a6.5 6.5 0 11-1.985-12.808A8.5 8.5 0 0135.5 15z"
            fill="url(#rmc-vant-empty-default__b)"
            opacity="0.8"
          />
          <path
            d="M133 34h18v14h8v83h-26V34z"
            fill="url(#rmc-vant-empty-default__c)"
            opacity="0.8"
          />
          <path
            d="M95 12h22.231v14H126v63.5H95V12z"
            fill="url(#rmc-vant-empty-default__d)"
            opacity="0.8"
          />
          <ellipse
            cx="79.961"
            cy="140.5"
            fill="url(#rmc-vant-empty-default__e)"
            fill-rule="nonzero"
            opacity="0.8"
            rx="45.961"
            ry="7.5"
          />
          <g
            fill-rule="nonzero"
            opacity="0.6"
          >
            <path
              d="M6.46 3.51l4.183 15.225A1 1 0 019.68 20H1.312a1 1 0 01-.964-1.265L4.53 3.51a1 1 0 011.929 0z"
              fill="url(#rmc-vant-empty-default__f)"
              transform="translate(134 115)"
            />
            <path
              d="M3.997 20h2.997v10H3.997z"
              fill="url(#rmc-vant-empty-default__g)"
              transform="translate(134 115)"
            />
            <path
              d="M6.46 3.51l4.183 15.225A1 1 0 019.68 20H1.312a1 1 0 01-.964-1.265L4.53 3.51a1 1 0 011.929 0z"
              fill="url(#rmc-vant-empty-default__f)"
              transform="translate(15 115)"
            />
            <path
              d="M3.997 20h2.997v10H3.997z"
              fill="url(#rmc-vant-empty-default__g)"
              transform="translate(15 115)"
            />
          </g>
          <g
            transform="translate(49 54)"
          >
            <mask
              fill="#fff"
              id="rmc-vant-empty-default__k"
            >
              <use
                xlink:href="#rmc-vant-empty-default__h"
              />
            </mask>
            <use
              fill="url(#rmc-vant-empty-default__i)"
              fill-rule="nonzero"
              xlink:href="#rmc-vant-empty-default__h"
            />
            <rect
              fill="url(#rmc-vant-empty-default__j)"
              fill-rule="nonzero"
              height="50"
              mask="url(#rmc-vant-empty-default__k)"
              opacity="0.4"
              rx="2"
              width="78"
            />
          </g>
          <path
            d="M0 14.367a2 2 0 012-2h74a2 2 0 012 2v46a2 2 0 01-2 2H10.632l-8.925 8.926a1 1 0 01-.576.284L1 71.586a1 1 0 01-1-1V14.367z"
            fill="url(#rmc-vant-empty-default__l)"
            fill-rule="nonzero"
            transform="translate(34 54)"
          />
          <g
            fill-rule="nonzero"
            opacity="0.8"
            transform="translate(48 77)"
          >
            <rect
              fill="url(#rmc-vant-empty-default__m)"
              height="6"
              rx="1"
              width="51"
            />
            <rect
              fill="url(#rmc-vant-empty-default__n)"
              height="6"
              rx="1"
              width="36"
              y="12"
            />
            <rect
              fill="url(#rmc-vant-empty-default__o)"
              height="6"
              rx="1"
              width="50"
              y="23"
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      class="rmcvant-empty-children"
    />
  </div>
</DocumentFragment>
`;
